### PR TITLE
Update GSP forecast route

### LINKED ISSUE #222: Update gsp/forecast/{gsp_id} route
## Detailed Description
We can update for `gsp/forecast/{gsp_id}` route to set `only_forecast_values=True`. Will need to update the documentation with this. 

## Context
- https://github.com/openclimatefix/uk-pv-national-gsp-api/issues/200, check that no users are using the soon to be removed routes

## Possible Implementation
- remove parameters `historic` and `only_forecast_values` and assume `only_forecast_values=True`
- add extra route which is `gsp/{gsp_id}/forecast` and point old route to it
- Update documentation

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

## Description
This PR does the following: 

- tidies up the GSP forecast API routes so they're more RESTful
- updates tests for the routes and removes unnecessary or duplicate tests
- updates the API documentation for the GSP forecast routes (see screenshots below)

`Get All Available Forecasts`
Screenshot 2023-06-28 at 11 00 42

`Get Forecasts For a Specific GSP`

Screenshot 2023-06-28 at 11 01 24

`Get Truths for All GSPs`
Screenshot 2023-06-28 at 11 02 44

`Get Truths for a Specific GSP`
Screenshot 2023-06-28 at 11 02 59

`Get GSP Boundaries `

Screenshot 2023-06-27 at 15 02 22

`Get System Details`

Screenshot 2023-06-27 at 15 04 37

Fixes #222

## How Has This Been Tested?

Tests were run with Docker and the documentation was checked by running the code locally. 

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/tests/test_gsp.py [hunk 1]
```diff
@@ -20,18 +20,18 @@
 
 @freeze_time("2022-01-01")
 def test_read_latest_one_gsp(db_session, api_client):
```
Comment (rachel-labri-tipton) [inline src/tests/test_gsp.py:22]: hi @peterdudfield, this PR isn't quite ready for review, but I wanted to ask a question about two tests. This one `test_read_latest_one_gsp` throws an error that the index is out of range. 

do I need this test now that the `/gsp/{gsp}/forecast` route has been updated? 

I'll leave my other question on the other problematic test.

File: src/tests/test_merged_routes.py [hunk 2]
```diff
@@ -12,41 +12,22 @@
 
 @freeze_time("2022-01-01")
 def test_read_one_gsp(db_session, api_client):
```
Comment (rachel-labri-tipton) [inline src/tests/test_merged_routes.py:14]: hello @peterdudfield, here's my second question about the tests for the gsp forecast route. `test_read_one_gsp` here also throws an error, and I'm not sure of the purpose of the test. And my code is wrong with `_ = ForecastValue(**response.json())`.  

Thanks in advance for any guidance.

File: src/tests/test_gsp.py (hunk 1 above)
Comment (peterdudfield) [reply src/tests/test_gsp.py:22]: interesteing, it shouldnt get an error. Whats is the error message?

Comment (peterdudfield) [reply src/tests/test_gsp.py:22]: I think ive fixed it

Comment (rachel-labri-tipton) [reply src/tests/test_gsp.py:22]: this makes sense now. the error message was that there was no object to concatenate.

**Review by peterdudfield — APPROVED** [review]

**Pull request merged**